js/actions/index.js

```diff
@@ -1,25 +1,58 @@
 // Action types to identify an action
 export const types = {
+  AGENCY_FINDER_DATA_FETCH: 'AGENCY_FINDER_DATA_FETCH',
+  AGENCY_FINDER_DATA_RECEIVE: 'AGENCY_FINDER_DATA_RECEIVE',
   REQUEST_AGENCY_CHANGE: 'REQUEST_AGENCY_CHANGE',
+  REQUEST_AGENCY_FETCH: 'REQUEST_AGENCY_FETCH',
   REQUEST_RECEIVE_AGENCY: 'REQUEST_RECEIVE_AGENCY',
 };
 
 // Action creators, to dispatch actions
-export function RequestActions({ dispatcher, api }) {
+export function RequestActions({ dispatcher, api, jsonapi }) {
   return {
+    fetchAgencyFinderData() {
+      dispatcher.dispatch({
+        type: types.AGENCY_FINDER_DATA_FETCH,
+      });
+
+      return jsonapi.params()
+        .include('agency')
+        .fields('agency', ['name', 'abbreviation'])
+        .fields('agency_component', ['title', 'abbreviation', 'agency'])
+        .limit(500) // TODO fetch over multiple requests
+        .get('/agency_components');
+    },
+
+    receiveAgencyFinderData(agencyComponents) {
+      dispatcher.dispatch({
+        type: types.AGENCY_FINDER_DATA_RECEIVE,
+        agencyComponents,
+      });
+
+      return Promise.resolve(agencyComponents);
+    },
+
     agencyChange(agency) {
       dispatcher.dispatch({
         type: types.REQUEST_AGENCY_CHANGE,
         agency,
       });
 
-      if (!agency) {
-        // The agency was unselected
-        return Promise.resolve(null);
+      return Promise.resolve(agency);
```

Comment:
Right, now once a selection is made in the agency finder, you'll be navigated to a different page. BUT this might change, I haven't seen the latest designs for this recently.